sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/Utility.java

```diff
@@ -388,86 +399,89 @@ public static ExtractKeyPhrasesResultCollection toExtractKeyPhrasesResultCollect
-//                            entityRelationMap.getOrDefault(targetEntity, new HashMap<>());
-//                        targetRelatedEntity.putIfAbsent(sourceEntity, relationType);
-//                        entityRelationMap.putIfAbsent(targetEntity, targetRelatedEntity);
-//
-//                        if (healthcareRelation.isBidirectional()) {
-//                            final Map<HealthcareEntity, HealthcareEntityRelationType> sourceRelatedEntity =
-//                                entityRelationMap.getOrDefault(sourceEntity, new HashMap<>());
-//                            sourceRelatedEntity.putIfAbsent(targetEntity, relationType);
-//                            entityRelationMap.putIfAbsent(sourceEntity, sourceRelatedEntity);
-//                        }
-                    });
-                }
-                healthcareEntities.forEach(healthcareEntity -> {
-                    if (entityRelationMap.containsKey(healthcareEntity)) {
-                        HealthcareEntityPropertiesHelper.setRelatedEntities(healthcareEntity,
-                            entityRelationMap.get(healthcareEntity));
-                    }
-                });
-
-                final AnalyzeHealthcareEntitiesResult analyzeHealthcareEntitiesResult =
-                    new AnalyzeHealthcareEntitiesResult(
-                        documentEntities.getId(),
-                        documentEntities.getStatistics() == null ? null
-                            : toTextDocumentStatistics(documentEntities.getStatistics()),
-                        null);
                 AnalyzeHealthcareEntitiesResultPropertiesHelper.setEntities(analyzeHealthcareEntitiesResult,
                     IterableStream.of(healthcareEntities));
-                AnalyzeHealthcareEntitiesResultPropertiesHelper.setWarnings(analyzeHealthcareEntitiesResult,
-                    IterableStream.of(warnings));
+
+                // Healthcare Entity relations
+                final List<HealthcareEntityRelation> healthcareEntityRelations =
+                    documentEntities.getRelations().stream().map(
+                        healthcareRelation -> {
+                            final HealthcareEntityRelation entityRelation = new HealthcareEntityRelation();
+                            // Healthcare relation type
+                            final RelationType relationType = healthcareRelation.getRelationType();
+                            if (relationType != null) {
+                                HealthcareEntityRelationPropertiesHelper.setRelationType(entityRelation,
+                                    HealthcareEntityRelationType.fromString(relationType.toString()));
+                            }
+                            // Healthcare entity relation roles
+                            final List<HealthcareEntityRelationRole> relationRoles =
+                                healthcareRelation.getEntities().stream().map(
```

Comment:
same here, can `getEntities()` be null?

Reply:
it can't be null since it is required in the swagger. https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.4/TextAnalytics.json#L1824